src/op_registry/onnx_registry.rs

```diff
@@ -0,0 +1,1192 @@
+impl_read_op!(Atan);
+impl_read_op!(AveragePool, |attrs: &Attrs| {
+    let PoolAttrs {
+        ceil_mode,
+        kernel_size,
+        padding,
+        strides,
+    } = get_common_pool_attrs(attrs);
+    let count_include_pad = attrs.get_as("count_include_pad").unwrap_or(false);
+
+    Ok(ops::AveragePool {
+        ceil_mode,
+        count_include_pad,
+        kernel_size,
+        padding,
+        strides,
+    })
+});
+impl_read_op!(BatchNormalization, |attrs: &Attrs| {
+    let epsilon = attrs.get("epsilon").map(|v| v.as_f32()).unwrap_or(1e-05);
+    Ok(ops::BatchNormalization { epsilon })
+});
+impl_read_op!(Cast, |attrs: &Attrs| {
+    let to = attrs.require("to")?.as_dtype()?;
+    Ok(ops::Cast { to })
+});
+impl_read_op!(CastLike);
+impl_read_op!(Ceil);
+impl_read_op!(Clip);
+impl_read_op!(Concat, |attrs: &Attrs| {
+    let axis = attrs.require("axis")?.as_i64() as isize;
+    Ok(ops::Concat { axis })
+});
+impl_read_op!(ConcatFromSequence, |attrs: &Attrs| {
+    let axis = attrs.require("axis")?.as_i64() as i32;
+    let new_axis = attrs.get("new_axis").map(|v| v.as_bool()).unwrap_or(false);
+    Ok(ops::ConcatFromSequence { axis, new_axis })
+});
+impl_read_op!(Conv, |attrs: &Attrs| {
+    attrs.get("kernel_shape"); // Ignored
+
+    let dilations = attrs
+        .get("dilations")
+        .map(|v| v.as_usize_ints())
+        .unwrap_or_default();
+    let pads = attrs
+        .get("pads")
+        .map(|v| v.as_usize_ints())
+        .unwrap_or_default();
+    let padding = Padding::Fixed(pads.into());
+    let groups = attrs.get("group").map(|v| v.as_i64()).unwrap_or(1);
+    let groups = groups as usize; // TODO - Handle invalid values
+    let strides = attrs
+        .get("strides")
+        .map(|v| v.as_usize_ints())
+        .unwrap_or_default();
+
+    Ok(ops::Conv {
+        dilations,
+        groups,
+        padding,
+        strides,
+    })
+});
+impl_read_op!(ConvInteger, |attrs: &Attrs| {
+    attrs.get("kernel_shape"); // Ignored
+
+    let dilations = attrs
+        .get("dilations")
+        .map(|v| v.as_usize_ints())
+        .unwrap_or_default();
+
+    // TODO - Padding should default to 0 along each spatial axis.
+    let pads = attrs
+        .get("pads")
+        .map(|v| v.as_usize_ints())
+        .unwrap_or_default();
+    let padding = Padding::Fixed(pads.into());
+
+    let groups = attrs.get("group").map(|v| v.as_i64()).unwrap_or(1);
+    let groups = groups as usize; // TODO - Handle invalid values
+
+    // TODO - This should default to [1] x spatial axis size.
+    let strides = attrs
+        .get("strides")
+        .map(|v| v.as_usize_ints())
+        .unwrap_or_default();
+
+    Ok(ops::ConvInteger {
+        dilations,
+        groups,
+        padding,
+        strides,
+    })
+});
+impl_read_op!(ConstantOfShape, |attrs: &Attrs| {
+    let value =
+        attrs
+            .get("value")
+            .map(|attr| {
+                let Some(tensor) = attr.attr.t.as_ref() else {
```

Comment:
TODO - This should re-use code from initializer deserialization in `onnx_loader.rs`.